src/main/java/appeng/recipes/handlers/GrinderHandler.java

```diff
@@ -0,0 +1,49 @@
+
+package appeng.recipes.handlers;
+
+
+import com.google.gson.JsonObject;
+
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.JsonUtils;
+import net.minecraftforge.common.crafting.CraftingHelper;
+import net.minecraftforge.common.crafting.JsonContext;
+
+import appeng.api.AEApi;
+import appeng.api.features.IGrinderRecipeBuilder;
+import appeng.api.features.IGrinderRegistry;
+import appeng.recipes.IAERecipeFactory;
+import appeng.recipes.factories.recipes.PartRecipeFactory;
+
+
+public class GrinderHandler implements IAERecipeFactory
+{
+
+	@Override
+	public void register( JsonObject json, JsonContext ctx )
+	{
+		// TODO only primary for now
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1135)